flocker/control/test/test_protocol.py

```diff
@@ -778,6 +815,69 @@ def test_second_configuration_change_waits_for_first_acknowledgement(self):
             ),
         )
 
+    def test_second_configuration_change_suceeds_when_first_fails(self):
+        """
+        If the first update fails, we want to ensure that following updates
+        won't get stuck waiting, and will get updated.
+        """
+
+        def raise_unexpected_exception(self,
+                                       commandType=None,
+                                       *a, **kw):
+            raise Exception("I'm an unexpected exception")
+
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
+        service = build_control_amp_service(self)
+        service.startService()
+
+        # Add a second agent, to ensure that the delayed logic interacts with
+        # the correct connection.
+        confounding_agent = FakeAgent()
+        confounding_client = AgentAMP(Clock(), confounding_agent)
+        failing_server = LoopbackAMPClient(confounding_client.locator)
+
+        # We want the update to fail in one of the connections
+        self.patch(failing_server,
+                   "callRemote",
+                   raise_unexpected_exception
+                   )
+        # The connection will fail, but it shouldn't prevent following
+        # commnads (from ``delayed_server``) to be properly executed
+        service.connected(failing_server)
+
+        configuration = service.configuration_service.get()
+        modified_configuration = arbitrary_transformation(configuration)
+
+        server = LoopbackAMPClient(client.locator)
+        delayed_server = DelayedAMPClient(server)
+        # Send first update
+        service.connected(delayed_server)
+        # Because the first agent fails to update, we do not
+        # expect the original configuration to change
+        first_agent_desired = configuration
```

Comment:
Later on in test you compare this to `configuration`. And it matches, unsurprisingly.

It seems like only thing you need to actually assert is that post-save() the second agent gets the modified_configuration? Which should fail before your fix and succeed after.